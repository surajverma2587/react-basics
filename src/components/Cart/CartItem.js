import React from 'react';
import PropTypes from 'prop-types';
import '../../assets/styles.css';

const CartItem = (/* props object */) => {
  /* bind the data from the props object to the html template */
  /* validate the props and render the component IF valid ELSE null */
  /* replace the bottom return statement with the populated html template */
  return (
    <div>
      CartItem Component
    </div>
  );
};

/* display name for the component */
CartItem.displayName = 'CartItem';

CartItem.propTypes = {
  /* add Proptypes here */
};

export default CartItem;