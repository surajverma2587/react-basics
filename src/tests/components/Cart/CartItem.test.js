import React from 'react';
import { shallow } from 'enzyme';

import CartItem from '../../../components/Cart/CartItem';

const props = {
  id: '1',
  title: 'Some title',
  price: 10,
  onClick: jest.fn()
};

describe('CartItem Component', () => {
  describe('when props are valid', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<CartItem {...props} />);
    });
    
    it('should render snapshot of the component', () => {
      expect(wrapper).toMatchSnapshot();
    });
  
    it('should render the component', () => {
      const cartItem = wrapper.find('.cart_item');
      expect(cartItem.exists()).toBe(true);
    });
  
    it('should render the title', () => {
      const title = wrapper.find('.cart_item--title');
      expect(title.text()).toEqual(props.title);
    });
  
    it('should render the price', () => {
      const price = wrapper.find('.cart_item--price');
      expect(price.text()).toEqual(`Â£${props.price}`);
    });
  
    it('should render the delete button', () => {
      const button = wrapper.find('button');
      expect(button.prop('id')).toEqual(props.id);
      expect(button.prop('onClick')).toEqual(props.onClick);
    });
  });
  
  describe('when props are invalid', () => {
    it('should not render the component when props are not present', () => {
      const wrapper = shallow(<CartItem />);
      expect(wrapper.find('.cart_item').exists()).toBe(false);
    });
  
    it('should not render the component when any prop is invalid', () => {
      const invalidProps = [
        { ...props, id: null },
        { ...props, title: null },
        { ...props, price: null },
        { ...props, onClick: null },
      ];
  
      invalidProps.forEach(invalidProp => {
        const wrapper = shallow(<CartItem {...invalidProp} />);
        expect(wrapper.find('.cart_item').exists()).toBe(false);
      });
    });
  });
});
