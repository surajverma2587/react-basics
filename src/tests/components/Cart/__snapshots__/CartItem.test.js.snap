// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem Component when props are valid should render snapshot of the component 1`] = `
<div
  className="cart_item"
>
  <div
    className="row"
  >
    <div
      className="col-sm-10 cart_item--title"
    >
      <span>
        Some title
      </span>
    </div>
    <div
      className="col-sm-1 cart_item--price"
    >
      <span>
        Â£
        10
      </span>
    </div>
    <div
      className="col-sm-1"
    >
      <button
        className="btn btn-danger"
        id="1"
        onClick={[MockFunction]}
        type="button"
      >
        -
      </button>
    </div>
  </div>
</div>
`;
